start.go

```diff
@@ -124,11 +135,30 @@ func startContainer(context *cli.Context, spec *specs.LinuxSpec, rspec *specs.Li
 	if err != nil {
 		return -1, err
 	}
-	handler := newSignalHandler(tty)
-	defer handler.Close()
 	if err := container.Start(process); err != nil {
 		return -1, err
 	}
+	if pidFile := context.String("pid-file"); pidFile != "" {
+		pid, err := process.Pid()
+		if err != nil {
+			return -1, err
+		}
+		f, err := os.Create(pidFile)
+		if err != nil {
+			logrus.WithField("pid", pid).Error("create pid file")
+		} else {
+			_, err = fmt.Fprintf(f, "%d", pid)
+			f.Close()
+			if err != nil {
+				logrus.WithField("error", err).Error("write pid file")
+			}
+		}
+	}
+	if detach {
+		return 0, nil
```

Comment:
what should happen to the stdin/stdout/stderr pipe of the process if we detach here ?